src/esp/metadata/managers/AttributesManagerBase.h

```diff
@@ -251,25 +251,32 @@ auto AttributesManager<T>::createFromJsonOrDefaultInternal(
     std::string& msg,
     bool registerObj) -> AttribsPtr {
   AttribsPtr attrs;
+  // Modify the passed filename to have the format of a legitimate configuration
+  // file for this Attributes by changing the extension
   std::string jsonAttrFileName =
-      this->convertFilenameToJSON(filename, this->JSONTypeExt_);
+      (Cr::Utility::String::endsWith(filename, this->JSONTypeExt_)
+           ? filename
+           : this->convertFilenameToJSON(filename, this->JSONTypeExt_));
+  // Check if this configuration file exists and if so use it to build
+  // attributes
   bool jsonFileExists = (this->isValidFileName(jsonAttrFileName));
   if (jsonFileExists) {
-    // check if stageAttributesHandle corresponds to an actual, existing
-    // json stage file descriptor.
-    // this method lives in class template.
+    // configuration file exists with requested name, use to build Attributes
     attrs = this->createObjectFromJSONFile(jsonAttrFileName, registerObj);
-    msg = "JSON File (" + jsonAttrFileName + ") Based";
+    msg = "JSON Configuration File (" + jsonAttrFileName + ") based";
   } else {
-    // if name is not json file descriptor but still appropriate file, or if
-    // is not a file .  Currently non-JSON files are treated as new, default
-    // attributes.
+    // An existing, valid configuration file could not be found using the passed
+    // filename.
+    // Currently non-JSON filenames are used to create new, default attributes.
     attrs = this->createDefaultObject(filename, registerObj);
+    // check if original filename is an actual object
     bool fileExists = (this->isValidFileName(filename));
+    // if filename passed is name of some kind of asset, or if it was not found
     if (fileExists) {
-      msg = "File (" + filename + ") not parsed, so new default";
+      msg = "File (" + filename +
+            ") not parseable configuration file, so new default";
     } else {
-      msg = "New default (" + filename + ")";
+      msg = filename + " file not found so new default";
```

Comment:
... so using new...

Reply:
Yes.  This is meant to inform the user that what they passed was not found in the file system (neither as a configuration file, nor as an appropriate asset that can be used to build a default configuration), so they were given a new one built from the specified default attributes' template.  This is meant to inform in case they perhaps spelled the name wrong, and wished to have one built from either a json file or an existing non-json asset.

Reply:
```suggestion
      msg = filename + " file not found so using new default";
```

Reply:
Actually that doesn't really work - this message is prepended to "<attributes type> created"
              << (registerTemplate ? " and registered." : ".");

Reply:
So for example, one feasible output is :

`8: I1109 16:33:15.267932 14811 AbstractObjectAttributesManagerBase.h:180] new_default_attributes file not found so new default Stage attributes created.
`

Reply:
Sounds good to me. Ship it @jturner65